src/main/js/field/input/DynamicSelect.js

```diff
@@ -53,12 +54,13 @@ function DynamicSelectInput(props) {
             removeSelected={removeSelected}
             options={options}
             placeholder={placeholder}
-            value={value}
+            value={selectValue}
             isMulti={multiSelect}
             closeMenuOnSelect={!multiSelect}
             components={components}
             isDisabled={readOnly}
             noOptionsMessage={() => 'No options available'}
+            onFocus={onFocus}
```

Comment:
I'm not sold on only taking action onFocus. It seems like a hack.

Reply:
If we were to not use AsyncSelect, what would be the better option here? We don't want to be making requests every single state change and this seems like a clean way to resolve this.

Reply:
If we do not use the AsyncSelect we need to use an event to initiate a request to fetch data.  onFocus is one event on the component that could be used.  The other is potentially onClick.  Ideally if we can add an onClick handler to the button used to display the list of options to select that would work better.

Reply:
After looking at this https://react-select.com/props, I am not sure that we have the option of an onClick for the drop down itself. Only for the options of the drop down. So I think onFocus would still be preferred.